src/marker/index.es

```diff
@@ -82,30 +85,24 @@ class Marker extends MapComponent {
     if (onPopupClose) marker.on('popupclose', onPopupClose);
     if (onClick) marker.on('click', onClick);
 
-    context.map.element.addLayer(marker);
+    this.getMap().element.addLayer(marker);
     this.setMarker(marker);
   }
 
   destroy() {
-    this.context.map.element.removeLayer(this.marker);
+    this.getMap().element.removeLayer(this.marker);
     this.marker.off();
     this.setMarker(null);
   }
 
   render() {
-    if (this.state.isReady) return this.props.children;
-    return null;
+    let content;
+    if (this.state.isReady) content = this.props.children;
+
+    return <Provider value={this.staticContext}>{content}</Provider>;
```

Comment:
We need to render `Provider` conditionally, so just return `null` if component is not ready, otherwise render `Provider` and `children`